### PR TITLE
Add some more tests and checks

### PR DESCRIPTION
**Pull request by Yash-10**
This pull request is a continuation of addition of some tests and checks.

- The altitude checks were added to prevent creation of an orbit with negative altitude. This might yield unexpected results at some places, for example, in the lithobrake event.

**Questions**

- It looks like `@pytest.mark.skipif("czml3" not in sys.modules, reason="requires czml3")` gets repeated for each test in `test_czml.py`. Could we instead have a global `pytestmark` to skip the whole module if `czml3` is not installed?

```python
pytestmark = pytest.mark.skipif("czml3" not in sys.modules)
```

- `ALL_PROPAGATORS` in `twobody/propagation.py`  seem to not contain `markley` and `danby`. This seems to be contrary to the name of the variable. Is this expected? If not, should we include all the propagators in `ALL_PROPAGATORS`?

- In #1253, you had advised to make `expected_doc` as a fixture. It seems that there are more `expected_doc`'s inside the tests. Is it necessary to have a fixture for each of them?

@astrojuanlu Could you let me know what do you think about the above questions?

Thanks!

### CODE REVIEW

**Comment by jorgepiloto** [comment]
The logic when building the `ALL_PROPAGATORS` macros is wrong! 😱

We are only including the propagators which are shared among the elliptic, parabolic and hyperbolic ones. Thank you for noticing this @Yash-10! 🚀

**Comment by jorgepiloto** [comment]
This could be fixed by doing:

```python
ALL_PROPAGATORS = set(ELLIPTIC_PROPAGATORS + PARABOLIC_PROPAGATORS + HYPERBOLIC_PROPAGATORS)
```

**Comment by Yash-10** [comment]
Thanks @jorgepiloto for your answer! Also, while we were on that module, I noticed that the "Note" inside `cowell`'s docstring mentioned `poliastro.integrators`. I thought it was removed some time back and I hence changed the docstring to mention scipy's `solve_ivp`.

File: src/poliastro/twobody/propagation.py
Comment (jorgepiloto) [inline src/poliastro/twobody/propagation.py:79]: I would also say in here the following:

```
... propagates to the maximum value (unless a terminal event is defined) and calculates ...
```

File: tests/tests_twobody/test_propagation.py
Comment (jorgepiloto) [inline tests/tests_twobody/test_propagation.py:469]: This is not required, there is a test already checking for `ALL_PROPAGATORS`, see:

https://github.com/poliastro/poliastro/blob/0a0c00559a36ce9779f824738773bc7e98a6697c/tests/tests_twobody/test_propagation.py#L105-L106

**Review by jorgepiloto — CHANGES_REQUESTED** [review]
Just some suggestions. After those, this is ready to be merged by my side 👍🏽

File: tests/tests_twobody/test_propagation.py
Comment (Yash-10) [reply tests/tests_twobody/test_propagation.py:469]: The motivation for adding this test was this line: https://github.com/poliastro/poliastro/blob/main/src/poliastro/twobody/propagation.py#L476. I noticed that below the "TODO"'s, there was an assert condition inside the source code. I though it would be good to turn it into its own test.  Also, this method uses the propagator method directly and not the propagation method of an orbit. Hence, after adding this test, the assert condition in `src` was removed.

What do you think about adding this test?

File: src/poliastro/twobody/propagation.py
Comment (Yash-10) [reply src/poliastro/twobody/propagation.py]: Thanks for this!

**Comment by astrojuanlu** [comment]
About having a global marker: yes, absolutely! I didn't know having a `pytestmark` would "apply marks at the module level" https://docs.pytest.org/en/6.2.x/example/markers.html#marking-whole-classes-or-modules

In #1253 I suggested to make `expected_doc` a fixture in case we wanted to reuse it for more than 1 test. If not, every test can have their own `expected_doc` and that's fine.

File: tests/tests_twobody/test_propagation.py
Comment (astrojuanlu) [reply tests/tests_twobody/test_propagation.py:469]: I'm +1 on adding this test and removing the `assert`!

**Review by astrojuanlu — APPROVED** [review]
This looks good to me! Ready to merge, unless @Yash-10 wants to make some extra additions (like the global pytest marker)

**Comment by astrojuanlu** [comment]
(And pending on the comment @jorgepiloto left)

**Comment by Yash-10** [comment]
Thanks @astrojuanlu I added the global pytest marker.

File: tests/tests_twobody/test_propagation.py
Comment (jorgepiloto) [reply tests/tests_twobody/test_propagation.py:469]: Oh, I see. Thanks for the clarification @Yash-10!

**Review by jorgepiloto — APPROVED** [review]
All modifications looking good, approving this 👍🏽

**Pull request merged**
